ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -770,9 +564,8 @@ ledgerDbSwitch :: (ApplyBlock l b, ResolvesBlocks r b m, c)
                -> [Ap m l r b c]  -- ^ New blocks to apply
                -> LedgerDB l r
                -> m (Either ExceededRollback (LedgerDB l r))
```

Comment:
The `ResolvesBlocks r b m` constraint on `ledgerDbSwitch` should go, it was only required to resolve blocks when rolling back, which should no longer be necessary. Note the `c` argument, which can still imply `ResolvesBlocks r b m` when the list of `Ap`s contains a `Re/ApplyRef`.